### PR TITLE
add IMAGE_CHANNEL_COUNT class variable to config

### PR DESCRIPTION
**Pull request by moorage**
Adds IMAGE_CHANNEL_COUNT class variable to config to make it easier to use Mask_RCNN for non 3-channel images, like grayscale or RGB-D

### CODE REVIEW

**Comment by keineahnung2345** [comment]
One thing to be noted: When IMAGE_CHANNEL_COUNT＝1, MEAN_PIXEL should be a single value.

**Comment by keineahnung2345** [comment]
@maxfrei750 Brilliant!

File: mrcnn/config.py [hunk 1]
```diff
@@ -210,9 +213,11 @@ def __init__(self):
 
         # Input image size
         if self.IMAGE_RESIZE_MODE == "crop":
-            self.IMAGE_SHAPE = np.array([self.IMAGE_MIN_DIM, self.IMAGE_MIN_DIM, 3])
+            self.IMAGE_SHAPE = np.array([self.IMAGE_MIN_DIM, self.IMAGE_MIN_DIM,
+                self.__class__.IMAGE_CHANNEL_COUNT])
         else:
-            self.IMAGE_SHAPE = np.array([self.IMAGE_MAX_DIM, self.IMAGE_MAX_DIM, 3])
+            self.IMAGE_SHAPE = np.array([self.IMAGE_MAX_DIM, self.IMAGE_MAX_DIM,
+                self.__class__.IMAGE_CHANNEL_COUNT])
```
Comment (waleedka) [inline mrcnn/config.py]: Is there a particular advantage to using self.__class__.IMAGE_CHANNEL_COUNT vs self.IMAGE_CHANNEL_COUNT?

**Comment by ApoorvaSuresh** [comment]
@lunasdejavu I am trying to work on 4 channel images, did you find a solution for including "conv1"?

**Pull request merged**